src/jrd/tra.cpp

```diff
@@ -143,7 +143,13 @@ void TRA_attach_request(Jrd::jrd_tra* transaction, Jrd::jrd_req* request)
 void TRA_detach_request(Jrd::jrd_req* request)
 {
 	if (!request->req_transaction)
+	{
+		fb_assert(!request->req_savepoints);
 		return;
+	}
+
+	// Release stored looper savepoints
+	Savepoint::destroy(request->req_savepoints);
```

Comment:
On 06/14/2016 01:13 PM, Dimitry Sibiryakov wrote:
>>   		return;
>> +	}
>> +
>> +	// Release stored looper savepoints
>> +	Savepoint::destroy(request->req_savepoints);
> This line causes engine to crash on test bugs.core_0214 as if savepoint was released from non-existing pool, but I cannot understand how it can be: pointers to transaction and pool are valid.

Always good idea to provide stack trace in such cases.